src/components/chip/selectable-chip/selectable-chip.test.tsx

```diff
@@ -0,0 +1,112 @@
+import { render, screen } from 'test-utils'
+import userEvent from '@testing-library/user-event'
+import SelectableChip from '.'
+import { LabelPairedAndroidIcon } from '@deriv/quill-icons'
+import { ChipSize } from '../types'
+
+describe('Selectable Chip', () => {
+  it('should handle onClick event', async () => {
+    const onClickButton = jest.fn()
+    render(
+      <SelectableChip size={'sm'} onClick={onClickButton}>
+        Selectbale Chip
+      </SelectableChip>,
+    )
+    const button = screen.getByRole('button', { name: /Selectbale Chip/i })
+    await userEvent.click(button)
+    expect(onClickButton).toBeCalled()
+  })
+
+  it('should not handle onClick event when it is disabled', async () => {
+    const onClickButton = jest.fn()
+    render(
+      <SelectableChip size={'sm'} disabled>
+        Selectbale Chip
+      </SelectableChip>,
+    )
+    const button = screen.getByRole('button', { name: /Selectbale Chip/i })
+    await userEvent.click(button)
+    expect(button).toBeDisabled()
+    expect(onClickButton).not.toBeCalled()
+  })
+
+  it('should toggle selected state on clicking the chip', async () => {
+    render(<SelectableChip size={'sm'}>Selectbale Chip</SelectableChip>)
+    const button = screen.getByRole('button', { name: /Selectbale Chip/i })
+    await userEvent.click(button)
+    expect(button).toHaveAttribute('data-state', 'selected')
+    await userEvent.click(button)
+    expect(button).toHaveAttribute('data-state', '')
+  })
+
+  it('should render correct disabled classname when it is disabled', () => {
+    render(<SelectableChip size={'sm'}>Selectbale Chip</SelectableChip>)
+    const button = screen.getByRole('button', { name: /Selectbale Chip/i })
+    expect(button).toHaveClass('disabled:opacity-600')
+  })
+
+  it('should render correct labelTag', () => {
+    render(
+      <SelectableChip size={'sm'} labelTag="labelTag">
+        Selectbale Chip
```

Comment:
**NIT**: I think u meant Selectable Chip as the innerText of <SelectableChip/>?

Reply:
Yes! Thanks for the catch! done ✅